src/test/fuzz/txdownloadman.cpp

```diff
@@ -0,0 +1,423 @@
+            },
+            [&] {
+                txdownloadman.GetRequestsToSend(rand_peer, time);
+            },
+            [&] {
+                txdownloadman.ReceivedTx(rand_peer, rand_tx);
+                const auto& [should_validate, maybe_package] = txdownloadman.ReceivedTx(rand_peer, rand_tx);
+                // The only possible results should be:
+                // - Don't validate the tx, no package.
+                // - Don't validate the tx, package.
+                // - Validate the tx, no package.
+                // The only combination that doesn't make sense is validate both tx and package.
+                Assert(!(should_validate && maybe_package.has_value()));
+                if (maybe_package.has_value()) CheckPackageToValidate(*maybe_package, rand_peer);
+            },
+            [&] {
+                txdownloadman.ReceivedNotFound(rand_peer, {rand_tx->GetWitnessHash()});
+            },
+            [&] {
+                const bool expect_work{txdownloadman.HaveMoreWork(rand_peer)};
+                const auto ptx = txdownloadman.GetTxToReconsider(rand_peer);
+                // expect_work=true doesn't necessarily mean the next item from the workset isn't a
+                // nullptr, as the transaction could have been removed from orphanage without being
+                // removed from the peer's workset.
+                if (ptx) {
+                    // However, if there was a non-null tx in the workset, HaveMoreWork should have
+                    // returned true.
+                    Assert(expect_work);
+                }
+            }
+        );
+        time += fuzzed_data_provider.PickValueInArray(TIME_SKIPS);
+    }
+    // Disconnect everybody, check that all data structures are empty.
+    for (NodeId nodeid = 0; nodeid <= NUM_PEERS; ++nodeid) {
+        txdownloadman.DisconnectedPeer(nodeid);
+        txdownloadman.CheckIsEmpty(nodeid);
+    }
+    txdownloadman.CheckIsEmpty();
+}
+
+// Give node 0 relay permissions, and nobody else. This helps us remember who is a RelayPermissions
+// peer without tracking anything (this is only for the txdownload_impl target).
+static bool HasRelayPermissions(NodeId peer) { return peer == 0; }
+
+static void CheckInvariants(const node::TxDownloadImpl& txdownload_impl, size_t max_orphan_count)
+{
+    const TxOrphanage& orphanage = txdownload_impl.m_orphanage;
+
+    // Orphanage usage should never exceed what is allowed
+    Assert(orphanage.Size() <= max_orphan_count);
+
+    // We should never have more than the maximum in-flight requests out for a peer.
+    for (NodeId peer = 0; peer < NUM_PEERS; ++peer) {
+        if (!HasRelayPermissions(peer)) {
+            Assert(txdownload_impl.m_txrequest.CountInFlight(peer) <= node::MAX_PEER_TX_REQUEST_IN_FLIGHT);
+        }
+    }
+    txdownload_impl.m_txrequest.SanityCheck();
+}
+
+FUZZ_TARGET(txdownload_impl, .init = initialize)
```

Comment:
```suggestion
FUZZ_TARGET(txdownloadman_impl, .init = initialize)
```

(if you take the other naming suggestion)

👉👈